### PR TITLE
ADR for Model-as-a-Service

### PR DESCRIPTION
**Pull request by TaoChenOSU**
### Motivation and Context

Related to: https://github.com/microsoft/semantic-kernel/issues/6530, https://github.com/microsoft/semantic-kernel/issues/3992

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: docs/decisions/0046-model-as-a-service.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+- Azure
+- AzureAI
+- AzureAIInference
+- AzureAIModelInference
+
+### Support for model-specific parameters
+
+Models can possess supplementary parameters that are not part of the default API. The service API and the client SDK enable the provision of model-specific parameters. Users can provide model-specific settings along with other settings, such as `temperature` and `top_p`, among others.
+
+In the context of SK, execution parameters are categorized under `PromptExecutionSettings`, which is inherited by all connector-specific setting classes. The settings of the new connector will contain a member of type `dictionary`, which will group together the model-specific parameters.
```
Comment (eavanvalkenburg) [inline docs/decisions/0046-model-as-a-service.md:44]: as discussed, no need to change the PromptExecutionSettings itself, create a subclass with set fields or do something with a dict inside of it, Hugging Face and Ollama are examples of this as well

File: docs/decisions/0046-model-as-a-service.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+At present, there is no official support for MaaS in SK. The purpose of this ADR is to examine the constraints of the service and explore potential solutions to enable support for the service in SK via the development of a new AI connector.
+
+## Client SDK
+
+The Azure team will be providing a new client library, namely `Azure.AI.Inference` in .Net and `azure-ai-inference` in Python, for effectively interacting with the service. While the service API is OpenAI-compatible, it is not permissible to use the OpenAI and the Azure OpenAI client libraries for interacting with the service as they are not independent with respect to both the models and their providers. This is because Azure AI Studio features a diverse range of open-source models, other than OpenAI models.
+
+### Limitations
+
+The initial release of the client SDK will only support chat completion and text/image embedding generation, with image generation to be added later.
```
Comment (eavanvalkenburg) [inline docs/decisions/0046-model-as-a-service.md:25]: image generation in general is not present in python yet. Image input is, will these models support that?

File: docs/decisions/0046-model-as-a-service.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+Plans to support for text completion are currently unclear, and it is highly unlikely that the SDK will ever include support for text completion.
```
Comment (eavanvalkenburg) [inline docs/decisions/0046-model-as-a-service.md:27]: Text completions are mentioned here: https://learn.microsoft.com/en-us/azure/ai-studio/reference/reference-model-inference-completions are they not available in the python package?

File: docs/decisions/0046-model-as-a-service.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+# These are optional elements. Feel free to remove any of them.
+status: { proposed }
+contact: { rogerbarreto, taochen }
+date: { 2024-06-12 }
+deciders: {}
+consulted: {}
+informed: {}
+---
+
+# Support for Model-as-a-Service in SK
```
Comment (eavanvalkenburg) [inline docs/decisions/0046-model-as-a-service.md:11]: should be Azure Model-as-a-Service

File: docs/decisions/0046-model-as-a-service.md (hunk 2 above)
Comment (TaoChenOSU) [reply docs/decisions/0046-model-as-a-service.md:25]: Yes, some of the models in the catalog support image inputs and the client SDKs also support image content as input.

File: docs/decisions/0046-model-as-a-service.md (hunk 3 above)
Comment (TaoChenOSU) [reply docs/decisions/0046-model-as-a-service.md:27]: Correct, the Python package won't have text completion. I believe the dotnet one also will not.

File: docs/decisions/0046-model-as-a-service.md (hunk 1 above)
Comment (TaoChenOSU) [reply docs/decisions/0046-model-as-a-service.md:44]: Correct, no need to change the base class.

File: docs/decisions/0046-azure-model-as-a-service.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+The Azure team will be providing a new client library, namely `Azure.AI.Inference` in .Net and `azure-ai-inference` in Python, for effectively interacting with the service. While the service API is OpenAI-compatible, it is not permissible to use the OpenAI and the Azure OpenAI client libraries for interacting with the service as they are not independent with respect to both the models and their providers. This is because Azure AI Studio features a diverse range of open-source models, other than OpenAI models.
+
+### Limitations
+
+The initial release of the client SDK will only support chat completion and text/image embedding generation, with image generation to be added later.
+
+Plans to support for text completion are currently unclear, and it is highly unlikely that the SDK will ever include support for text completion.
+
+TODO: Should we add support for text completion in SK by skipping the client SDK and talk directly to the REST API? This is not a great option. Given that text completion is one of the foundation AI services SK provides, we should get more inputs.
```
Comment (alliscode) [inline docs/decisions/0046-azure-model-as-a-service.md:29]: For at least the initial version of this, let's not add our own functionality for text completions. If we get a lot of feedback about it, then we could reassess.

**Review by alliscode — APPROVED** [review]

**Review by moonbox3 — APPROVED** [review]
LGTM

**Review by RogerBarreto — APPROVED** [review]

**Comment by KSemenenko** [comment]
Is this available for c#?

**Pull request merged**